i18n/management/commands/publish_i18n.py

```diff
@@ -23,46 +35,60 @@ class Command(BaseCommand):
 
     def __init__(self, *args, **kwargs):
         self.total_elapsed_time = 0
-        self.total_pdf_generation_time = 0
+        self.pdf_publishing_time = 0
+        self.html_publishing_time = 0
+        self.pool = multiprocessing.Pool(multiprocessing.cpu_count())
 
         with open(CHANGES_JSON) as changes_json:
             self.changes = json.load(changes_json)
 
         super(Command, self).__init__(*args, **kwargs)
 
+    def get_language_codes_with_changes(self):
+        return [language_code for language_code
+                in get_non_english_language_codes()
+                if self.has_changes(language_code)]
+
     def has_changes(self, language_code):
-        # Because the published contents of a model include contents from other
-        # models, we can't easily determine whether a specific model should be
-        # published or not based on the changed files. So, we just determine
-        # changes on a per-language (rather than per-language and also
-        # per-model) basis. This should be sufficient for our needs given the
-        # relatively low translation activity for this project, but we may need
-        # to make this more sophisticated in the future if that changes.
-        return len(self.changes[language_code]) > 0
-
-    def publish_object(self, obj, language_code):
+        # Because the published contents of a model include contents from other models, we can't
+        # easily determine whether a specific model should be published or not based on the changed
+        # files. So, we just determine changes on a per-language (rather than per-language and also
+        # per-model) basis. This should be sufficient for our needs given the relatively low
+        # translation activity for this project, but we may need to make this more sophisticated in
+        # the future if that changes.
+        return len(self.changes.get(language_code, [])) > 0
+
+    def publish_objects_in_language(self, objects, language_code):
         translation.activate(language_code)
 
-        if hasattr(obj, 'publish'):
```

Comment:
Do we not need this check anymore?

Reply:
we do, it just moved! It's up here now https://github.com/code-dot-org/curriculumbuilder/pull/309/files#diff-84e2554603b1b28645da9724b3100e34R20